### PR TITLE
Engage pages and takeovers

### PR DESCRIPTION
**Pull request by carkod**
## Done
- Scss linting config consistent with ubuntu.com. so a lot of file formatting.
- Move engage pages to discourse
- Removed html files
- JS homepage takeovers

## QA
- Go to https://cn-ubuntu-com-475.demos.haus and check the homepage takeovers are correct
  - Active takeovers are showing
  - Image sizes are correct
  - They display even if you break the takeovers.json endpoint

- You can see the list of takeovers https://cn-ubuntu-com-475.demos.haus/takeovers
- Go to https://cn-ubuntu-com-475.demos.haus/engage
- Click on list of engage pages, make sure they show and forms work

Edit content: https://discourse.ubuntu.com/t/engage-pages-and-takeovers-for-cn-ubuntu-com/19117

## Reminder before merge
Replace thank-you page absolute links to `https://cn.ubuntu.com`

## Issue / Card

#452

### CODE REVIEW

**Comment by nottrobin** [comment]
I noticed the page jumps down on reload because of the global nav. This doesn't happen on ubuntu.com because there's a reservation for the global nav in the default styles:

image

(Try loading this demo vs ubuntu.com without JavaScript and with JavaScript to see what I mean).

Could you add this reservation at the top for cn.ubuntu.com as well to avoid the jumping?

**Comment by carkod** [comment]
> I noticed the page jumps down on reload because of the global nav. This doesn't happen on ubuntu.com because there's a reservation for the global nav in the default styles:
> 
> image
> 
> vs:
> 
> image
> 
> (Try loading this demo vs ubuntu.com without JavaScript and with JavaScript to see what I mean).
> 
> Could you add this reservation at the top for cn.ubuntu.com as well to avoid the jumping?

Fixed

**Comment by zprood** [comment]
It looks great! Thank you!

File: templates/engage/thank-you.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      {% if 'pages.ubuntu.com' or 'assets.ubuntu.com' in resource_url %}
+      <p>
+        你的{{ resource_name }}现已可下载... 
+      </p>
+      <p>
+        <a class="p-button--positive" href="{{ resource_url }}">立即下载</a>
+      </p>
+      {% else %}
+        <p>
+          抱歉，我们无法识别此下载请求。 请通过 <a class="p-link--external" href="https://github.com/canonical-websites/www.ubuntu.com/issues/new?body=Bad+resource+download+link+from {{ request.url | urlencode }}">GitHub</a> 反馈.
```
Comment (anthonydillon) [inline templates/engage/thank-you.html:44]: Could you update this link please

**Review by anthonydillon — APPROVED** [review]
LGTM :+1:

**Pull request merged**